diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 65e5ab6..a7cc4b6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1049,9 +1049,6 @@ if(BUILD_CLIENT AND NOT ANDROID)
 		ARCHIVE DESTINATION ${BINDIR}
 		BUNDLE DESTINATION .
 	)
-	if(UNIX)
-		CreateLegacyAlias(minetest_alias "${EXECUTABLE_OUTPUT_PATH}/minetest" ${PROJECT_NAME})
-	endif()
 
 	if(APPLE)
 		install(CODE "
@@ -1075,9 +1072,6 @@ endif()
 
 if(BUILD_SERVER)
 	install(TARGETS ${PROJECT_NAME}server DESTINATION ${BINDIR})
-	if(UNIX)
-		CreateLegacyAlias(minetestserver_alias "${EXECUTABLE_OUTPUT_PATH}/minetestserver" ${PROJECT_NAME}server)
-	endif()
 endif()
 
 if (ANDROID)
