diff --git a/sources/libClangSharp/CMakeLists.txt b/sources/libClangSharp/CMakeLists.txt
index 2c64cc1..f0dbc59 100644
--- a/sources/libClangSharp/CMakeLists.txt
+++ b/sources/libClangSharp/CMakeLists.txt
@@ -39,7 +39,11 @@ find_package(Clang REQUIRED CONFIG
 
 add_library(ClangSharp SHARED ${SOURCES})
 
-target_link_libraries(ClangSharp PRIVATE clangAST clangFrontend libclang)
+if (TARGET clang-cpp) # Linux packages install libclang and clang-cpp
+    target_link_libraries(ClangSharp PRIVATE clang-cpp libclang)
+else()
+    target_link_libraries(ClangSharp PRIVATE clangAST clangFrontend libclang)
+endif()
 
 target_include_directories(ClangSharp PRIVATE ${CLANG_INCLUDE_DIRS})
 
