diff --git a/externals/CMakeLists.txt b/externals/CMakeLists.txt
index 811836d..a79e72b 100644
--- a/externals/CMakeLists.txt
+++ b/externals/CMakeLists.txt
@@ -62,8 +62,8 @@ if (APPLE)
     endif()
 endif()
 
-add_subdirectory(volk)
+# add_subdirectory(volk)
 
 #nlohmann json 
-set(JSON_BuildTests OFF CACHE INTERNAL "")
-add_subdirectory(json)
+# set(JSON_BuildTests OFF CACHE INTERNAL "")
+# add_subdirectory(json)
