diff --git a/libraries/gamemode/include/gamemode_client.h b/libraries/gamemode/include/gamemode_client.h
index b186cd4..9ff0e92 100644
--- a/libraries/gamemode/include/gamemode_client.h
+++ b/libraries/gamemode/include/gamemode_client.h
@@ -168,17 +168,11 @@ __attribute__((always_inline)) static inline int internal_load_libgamemode(void)
     void* libgamemode = NULL;
 
     /* Try and load libgamemode */
-    libgamemode = dlopen("libgamemode.so.0", RTLD_NOW);
+    libgamemode = dlopen("libgamemode.so.0.0.0", RTLD_NOW);
     if (!libgamemode) {
-        /* Attempt to load unversioned library for compatibility with older
-         * versions (as of writing, there are no ABI changes between the two -
-         * this may need to change if ever ABI-breaking changes are made) */
-        libgamemode = dlopen("libgamemode.so", RTLD_NOW);
-        if (!libgamemode) {
-            snprintf(internal_gamemode_client_error_string, sizeof(internal_gamemode_client_error_string), "dlopen failed - %s", dlerror());
-            internal_libgamemode_loaded = -1;
-            return -1;
-        }
+        snprintf(internal_gamemode_client_error_string, sizeof(internal_gamemode_client_error_string), "dlopen failed - %s", dlerror());
+        internal_libgamemode_loaded = -1;
+        return -1;
     }
 
     /* Attempt to bind all symbols */
