diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3dabdb8..6d70e66 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -230,7 +230,7 @@ endif()
 find_package(PajladaSerialize REQUIRED)
 find_package(PajladaSignals REQUIRED)
 find_package(LRUCache REQUIRED)
-find_package(MagicEnum REQUIRED)
+find_package(magic_enum REQUIRED)
 find_package(Doxygen)
 find_package(BoostCertify REQUIRED)
 
diff --git a/src/common/Channel.hpp b/src/common/Channel.hpp
index 4c08362..5d5aaaa 100644
--- a/src/common/Channel.hpp
+++ b/src/common/Channel.hpp
@@ -6,7 +6,7 @@
 #include "messages/MessageFlag.hpp"
 #include "messages/MessageSink.hpp"
 
-#include <magic_enum/magic_enum.hpp>
+#include <magic_enum.hpp>
 #include <pajlada/signals/signal.hpp>
 #include <QDate>
 #include <QString>
diff --git a/src/common/network/NetworkPrivate.cpp b/src/common/network/NetworkPrivate.cpp
index 9261512..6c9fe2b 100644
--- a/src/common/network/NetworkPrivate.cpp
+++ b/src/common/network/NetworkPrivate.cpp
@@ -11,7 +11,7 @@
 #include "util/PostToThread.hpp"
 #include "util/QMagicEnum.hpp"
 
-#include <magic_enum/magic_enum.hpp>
+#include <magic_enum.hpp>
 #include <QCryptographicHash>
 #include <QElapsedTimer>
 #include <QFile>
diff --git a/src/controllers/plugins/LuaUtilities.hpp b/src/controllers/plugins/LuaUtilities.hpp
index 0f7bdc5..a6e5c62 100644
--- a/src/controllers/plugins/LuaUtilities.hpp
+++ b/src/controllers/plugins/LuaUtilities.hpp
@@ -4,7 +4,7 @@
 
 #    include <lua.h>
 #    include <lualib.h>
-#    include <magic_enum/magic_enum.hpp>
+#    include <magic_enum.hpp>
 #    include <QList>
 #    include <sol/state_view.hpp>
 
diff --git a/src/controllers/plugins/Plugin.cpp b/src/controllers/plugins/Plugin.cpp
index 739fa63..c07048d 100644
--- a/src/controllers/plugins/Plugin.cpp
+++ b/src/controllers/plugins/Plugin.cpp
@@ -8,7 +8,7 @@
 #    include "util/QMagicEnum.hpp"
 
 #    include <lua.h>
-#    include <magic_enum/magic_enum.hpp>
+#    include <magic_enum.hpp>
 #    include <QJsonArray>
 #    include <QJsonObject>
 #    include <QLoggingCategory>
diff --git a/src/controllers/plugins/PluginPermission.cpp b/src/controllers/plugins/PluginPermission.cpp
index 9a4bc92..56dbe37 100644
--- a/src/controllers/plugins/PluginPermission.cpp
+++ b/src/controllers/plugins/PluginPermission.cpp
@@ -3,7 +3,7 @@
 
 #    include "util/QMagicEnum.hpp"
 
-#    include <magic_enum/magic_enum.hpp>
+#    include <magic_enum.hpp>
 #    include <QJsonObject>
 
 namespace chatterino {
diff --git a/src/messages/MessageElement.hpp b/src/messages/MessageElement.hpp
index 12de603..e69c1ad 100644
--- a/src/messages/MessageElement.hpp
+++ b/src/messages/MessageElement.hpp
@@ -7,7 +7,7 @@
 #include "providers/links/LinkInfo.hpp"
 #include "singletons/Fonts.hpp"
 
-#include <magic_enum/magic_enum.hpp>
+#include <magic_enum.hpp>
 #include <pajlada/signals/signalholder.hpp>
 #include <QRect>
 #include <QString>
diff --git a/src/messages/MessageFlag.hpp b/src/messages/MessageFlag.hpp
index 48eacaf..cb0eb69 100644
--- a/src/messages/MessageFlag.hpp
+++ b/src/messages/MessageFlag.hpp
@@ -2,7 +2,7 @@
 
 #include "common/FlagsEnum.hpp"
 
-#include <magic_enum/magic_enum.hpp>
+#include <magic_enum.hpp>
 
 namespace chatterino {
 
diff --git a/src/providers/seventv/SeventvCosmetics.hpp b/src/providers/seventv/SeventvCosmetics.hpp
index 6302c51..0d521ac 100644
--- a/src/providers/seventv/SeventvCosmetics.hpp
+++ b/src/providers/seventv/SeventvCosmetics.hpp
@@ -1,6 +1,6 @@
 #pragma once
 
-#include <magic_enum/magic_enum.hpp>
+#include <magic_enum.hpp>
 
 namespace chatterino::seventv {
 
diff --git a/src/providers/seventv/eventapi/Message.hpp b/src/providers/seventv/eventapi/Message.hpp
index 5a3eebc..fca75eb 100644
--- a/src/providers/seventv/eventapi/Message.hpp
+++ b/src/providers/seventv/eventapi/Message.hpp
@@ -2,7 +2,7 @@
 
 #include "providers/seventv/eventapi/Subscription.hpp"
 
-#include <magic_enum/magic_enum.hpp>
+#include <magic_enum.hpp>
 #include <QJsonDocument>
 #include <QJsonObject>
 #include <QString>
diff --git a/src/providers/seventv/eventapi/Subscription.hpp b/src/providers/seventv/eventapi/Subscription.hpp
index c6767b1..691c99a 100644
--- a/src/providers/seventv/eventapi/Subscription.hpp
+++ b/src/providers/seventv/eventapi/Subscription.hpp
@@ -1,6 +1,6 @@
 #pragma once
 
-#include <magic_enum/magic_enum.hpp>
+#include <magic_enum.hpp>
 #include <QByteArray>
 #include <QHash>
 #include <QJsonObject>
diff --git a/src/providers/twitch/api/Helix.cpp b/src/providers/twitch/api/Helix.cpp
index 72ba887..0095c5f 100644
--- a/src/providers/twitch/api/Helix.cpp
+++ b/src/providers/twitch/api/Helix.cpp
@@ -9,7 +9,7 @@
 #include "util/CancellationToken.hpp"
 #include "util/QMagicEnum.hpp"
 
-#include <magic_enum/magic_enum.hpp>
+#include <magic_enum.hpp>
 #include <QJsonDocument>
 #include <QStringBuilder>
 
diff --git a/src/providers/twitch/pubsubmessages/AutoMod.hpp b/src/providers/twitch/pubsubmessages/AutoMod.hpp
index 4312d52..dc82a58 100644
--- a/src/providers/twitch/pubsubmessages/AutoMod.hpp
+++ b/src/providers/twitch/pubsubmessages/AutoMod.hpp
@@ -1,6 +1,6 @@
 #pragma once
 
-#include <magic_enum/magic_enum.hpp>
+#include <magic_enum.hpp>
 #include <QColor>
 #include <QJsonObject>
 #include <QString>
diff --git a/src/providers/twitch/pubsubmessages/Base.hpp b/src/providers/twitch/pubsubmessages/Base.hpp
index a1da168..a58b121 100644
--- a/src/providers/twitch/pubsubmessages/Base.hpp
+++ b/src/providers/twitch/pubsubmessages/Base.hpp
@@ -1,6 +1,6 @@
 #pragma once
 
-#include <magic_enum/magic_enum.hpp>
+#include <magic_enum.hpp>
 #include <QJsonDocument>
 #include <QJsonObject>
 #include <QString>
diff --git a/src/providers/twitch/pubsubmessages/ChannelPoints.hpp b/src/providers/twitch/pubsubmessages/ChannelPoints.hpp
index fe2254a..5c19584 100644
--- a/src/providers/twitch/pubsubmessages/ChannelPoints.hpp
+++ b/src/providers/twitch/pubsubmessages/ChannelPoints.hpp
@@ -1,6 +1,6 @@
 #pragma once
 
-#include <magic_enum/magic_enum.hpp>
+#include <magic_enum.hpp>
 #include <QJsonObject>
 #include <QString>
 
diff --git a/src/providers/twitch/pubsubmessages/ChatModeratorAction.hpp b/src/providers/twitch/pubsubmessages/ChatModeratorAction.hpp
index e04019c..5f29673 100644
--- a/src/providers/twitch/pubsubmessages/ChatModeratorAction.hpp
+++ b/src/providers/twitch/pubsubmessages/ChatModeratorAction.hpp
@@ -1,6 +1,6 @@
 #pragma once
 
-#include <magic_enum/magic_enum.hpp>
+#include <magic_enum.hpp>
 #include <QJsonObject>
 #include <QString>
 
diff --git a/src/providers/twitch/pubsubmessages/LowTrustUsers.hpp b/src/providers/twitch/pubsubmessages/LowTrustUsers.hpp
index e266628..cd9fb74 100644
--- a/src/providers/twitch/pubsubmessages/LowTrustUsers.hpp
+++ b/src/providers/twitch/pubsubmessages/LowTrustUsers.hpp
@@ -3,7 +3,7 @@
 #include "providers/twitch/TwitchBadge.hpp"
 
 #include <common/FlagsEnum.hpp>
-#include <magic_enum/magic_enum.hpp>
+#include <magic_enum.hpp>
 #include <QColor>
 #include <QJsonObject>
 #include <QString>
diff --git a/src/util/QMagicEnum.hpp b/src/util/QMagicEnum.hpp
index 2e1d7a7..c26d141 100644
--- a/src/util/QMagicEnum.hpp
+++ b/src/util/QMagicEnum.hpp
@@ -1,6 +1,6 @@
 #pragma once
 
-#include <magic_enum/magic_enum.hpp>
+#include <magic_enum.hpp>
 #include <QString>
 #include <QStringView>
 
diff --git a/src/widgets/helper/ChannelView.cpp b/src/widgets/helper/ChannelView.cpp
index b3e14ea..a07f0f9 100644
--- a/src/widgets/helper/ChannelView.cpp
+++ b/src/widgets/helper/ChannelView.cpp
@@ -47,7 +47,7 @@
 #include "widgets/TooltipWidget.hpp"
 #include "widgets/Window.hpp"
 
-#include <magic_enum/magic_enum_flags.hpp>
+#include <magic_enum_flags.hpp>
 #include <QApplication>
 #include <QClipboard>
 #include <QColor>
diff --git a/src/widgets/settingspages/GeneralPage.cpp b/src/widgets/settingspages/GeneralPage.cpp
index e1f045a..54fe784 100644
--- a/src/widgets/settingspages/GeneralPage.cpp
+++ b/src/widgets/settingspages/GeneralPage.cpp
@@ -22,7 +22,7 @@
 #include "widgets/settingspages/GeneralPageView.hpp"
 #include "widgets/settingspages/SettingWidget.hpp"
 
-#include <magic_enum/magic_enum.hpp>
+#include <magic_enum.hpp>
 #include <QDesktopServices>
 #include <QFileDialog>
 #include <QFontDialog>
diff --git a/tools/crash-handler/src/WinSupport.cpp b/tools/crash-handler/src/WinSupport.cpp
index c602496..ca1f4fa 100644
--- a/tools/crash-handler/src/WinSupport.cpp
+++ b/tools/crash-handler/src/WinSupport.cpp
@@ -6,7 +6,7 @@
 #define MAGIC_ENUM_USING_ALIAS_STRING_VIEW \
     using string_view = std::wstring_view;
 #define MAGIC_ENUM_USING_ALIAS_STRING using string = std::wstring;
-#include <magic_enum/magic_enum.hpp>
+#include <magic_enum.hpp>
 
 using namespace std::string_literals;
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 8f3fee2..b66fe89 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -861,7 +861,6 @@ target_link_libraries(${LIBRARY_PROJECT}
         Threads::Threads
         RapidJSON::RapidJSON
         LRUCache
-        MagicEnum
         $<$<BOOL:${WIN32}>:Wtsapi32>
         twitch-eventsub-ws
         BoostCertify
