diff --git a/scdl/scdl.py b/scdl/scdl.py
index f6a4f6d..97ba424 100644
--- a/scdl/scdl.py
+++ b/scdl/scdl.py
@@ -504,7 +504,7 @@ def get_config(config_file: pathlib.Path) -> configparser.ConfigParser:
     """Gets config from scdl.cfg"""
     config = configparser.ConfigParser()
 
-    default_config_file = pathlib.Path(__file__).with_name("scdl.cfg")
+    default_config_file = "/etc/scdl.cfg"
 
     with get_filelock(config_file):
         # load default config first
