diff --git a/intern/cycles/scene/osl.cpp b/intern/cycles/scene/osl.cpp
index eb0eb18feeb..13976748d9a 100644
--- a/intern/cycles/scene/osl.cpp
+++ b/intern/cycles/scene/osl.cpp
@@ -233,7 +233,7 @@ void OSLManager::texture_system_init()
 #  if OIIO_VERSION_MAJOR >= 3
     ts_shared = OSL::TextureSystem::create(false);
 #  else
-    ts_shared = shared_ptr(OSL::TextureSystem::create(false), OSL::TextureSystem::destroy);
+    ts_shared = std::shared_ptr<OSL::TextureSystem>(OSL::TextureSystem::create(false), [](OSL::TextureSystem* ptr) { OSL::TextureSystem::destroy(ptr); });
 #  endif
 
     ts_shared->attribute("automip", 1);
