diff --git a/build_files/cmake/macros.cmake b/build_files/cmake/macros.cmake
index 64b5c36843d..b8615500a34 100644
--- a/build_files/cmake/macros.cmake
+++ b/build_files/cmake/macros.cmake
@@ -881,7 +881,7 @@ function(get_blender_version)
   math(EXPR _out_version_minor "${_out_version} % 100")
 
   # output vars
-  set(BLENDER_VERSION "${_out_version_major}.${_out_version_minor}" PARENT_SCOPE)
+  set(BLENDER_VERSION "${_out_version_major}.${_out_version_minor}-__BLENDER_BRANCH__" PARENT_SCOPE)
   set(BLENDER_VERSION_MAJOR "${_out_version_major}" PARENT_SCOPE)
   set(BLENDER_VERSION_MINOR "${_out_version_minor}" PARENT_SCOPE)
   set(BLENDER_VERSION_PATCH "${_out_version_patch}" PARENT_SCOPE)
diff --git a/source/blender/blenkernel/intern/appdir.cc b/source/blender/blenkernel/intern/appdir.cc
index 179b1f61210..bd501d63a28 100644
--- a/source/blender/blenkernel/intern/appdir.cc
+++ b/source/blender/blenkernel/intern/appdir.cc
@@ -119,9 +119,13 @@ void BKE_appdir_exit()
  */
 static char *blender_version_decimal(const int version)
 {
-  static char version_str[5];
+  static char version_str[255];
   BLI_assert(version < 1000);
-  SNPRINTF(version_str, "%d.%d", version / 100, version % 100);
+  if (version != BLENDER_VERSION) {
+    SNPRINTF(version_str, "%d.%d", version / 100, version % 100);
+  } else {
+    SNPRINTF(version_str, "%d.%d-%s", version / 100, version % 100, "__BLENDER_BRANCH__");
+  }
   return version_str;
 }
 
