diff --git a/source/blender/draw/engines/overlay/overlay_next_private.hh b/source/blender/draw/engines/overlay/overlay_next_private.hh
index ea64646fc04..86edcd00ba8 100644
--- a/source/blender/draw/engines/overlay/overlay_next_private.hh
+++ b/source/blender/draw/engines/overlay/overlay_next_private.hh
@@ -8,6 +8,8 @@
 
 #pragma once
 
+#include <functional>
+
 #include "BKE_movieclip.h"
 #include "BKE_object.hh"
 
