diff --git a/otherarch/sdcpp/tae.hpp b/otherarch/sdcpp/tae.hpp
index 6830598..2e8d1b8 100644
--- a/otherarch/sdcpp/tae.hpp
+++ b/otherarch/sdcpp/tae.hpp
@@ -201,7 +201,7 @@ struct TinyAutoEncoder : public GGMLRunner {
                     bool decoder_only = true,
                     SDVersion version = VERSION_SD1)
         : decode_only(decoder_only),
-          taesd(decode_only, version),
+          taesd(decoder_only, version),
           GGMLRunner(backend) {
         taesd.init(params_ctx, tensor_types, prefix);
     }
