diff --git a/Makefile b/Makefile
index 40e2a67..ad5cf6a 100644
--- a/Makefile
+++ b/Makefile
@@ -54,8 +54,8 @@ endif
 CFLAGS   += -I. -Iggml/include -Iggml/src -Iggml/src/ggml-cpu -Iinclude -Isrc -I./include -I./include/CL -I./otherarch -I./otherarch/tools -I./otherarch/sdcpp -I./otherarch/sdcpp/thirdparty -I./include/vulkan -O3 -fno-finite-math-only -std=c11 -fPIC -DLOG_DISABLE_LOGS -D_GNU_SOURCE -DGGML_USE_CPU -DGGML_USE_CPU_AARCH64
 CXXFLAGS += -I. -Iggml/include -Iggml/src -Iggml/src/ggml-cpu -Iinclude -Isrc -I./common -I./include -I./include/CL -I./otherarch -I./otherarch/tools -I./otherarch/sdcpp -I./otherarch/sdcpp/thirdparty -I./include/vulkan -O3 -fno-finite-math-only -std=c++17 -fPIC -DLOG_DISABLE_LOGS -D_GNU_SOURCE -DGGML_USE_CPU -DGGML_USE_CPU_AARCH64
 ifndef KCPP_DEBUG
-	CFLAGS += -DNDEBUG -s
-	CXXFLAGS += -DNDEBUG -s
+	CFLAGS += -DNDEBUG
+	CXXFLAGS += -DNDEBUG
 endif
 ifdef LLAMA_NO_LLAMAFILE
 GGML_NO_LLAMAFILE := 1
