diff --git a/src/dynarmic/CMakeLists.txt b/src/dynarmic/CMakeLists.txt
index 6d19c43..b9e48e2 100644
--- a/src/dynarmic/CMakeLists.txt
+++ b/src/dynarmic/CMakeLists.txt
@@ -123,7 +123,10 @@ if ("riscv" IN_LIST ARCHITECTURE)
 endif()
 
 if ("x86_64" IN_LIST ARCHITECTURE)
-    find_package(xbyak 7 CONFIG)
+    add_library(xbyak INTERFACE)
+    add_library(xbyak::xbyak ALIAS xbyak)
+    target_compile_options(xbyak INTERFACE "-fno-operator-names")
+    target_compile_definitions(xbyak INTERFACE XBYAK_STRICT_CHECK_MEM_REG_SIZE=0)
 endif()
 
 if (DYNARMIC_USE_LLVM)
