diff --git a/src/dynarmic/CMakeLists.txt b/src/dynarmic/CMakeLists.txt
index e128f66..a705c02 100644
--- a/src/dynarmic/CMakeLists.txt
+++ b/src/dynarmic/CMakeLists.txt
@@ -136,9 +136,12 @@ if ("riscv" IN_LIST ARCHITECTURE)
 endif()
 
 if ("x86_64" IN_LIST ARCHITECTURE)
-    find_package(xbyak 7 CONFIG)
+    add_library(xbyak INTERFACE)
+    add_library(xbyak::xbyak ALIAS xbyak)
+    target_compile_options(xbyak INTERFACE "-fno-operator-names")
+    target_compile_definitions(xbyak INTERFACE XBYAK_STRICT_CHECK_MEM_REG_SIZE=0)
     find_package(zycore REQUIRED)
-    find_package(zydis 4 REQUIRED)
+    find_package(zydis REQUIRED)
 endif()
 
 if (DYNARMIC_USE_LLVM)
diff --git a/src/dynarmic/externals/cpmfile.json b/src/dynarmic/externals/cpmfile.json
index f1fd47d..ca0487f 100644
--- a/src/dynarmic/externals/cpmfile.json
+++ b/src/dynarmic/externals/cpmfile.json
@@ -26,7 +26,6 @@
     },
     "zydis": {
         "package": "zydis",
-        "version": "4",
         "repo": "zyantific/zydis",
         "sha": "c2d2bab025",
         "hash": "3808773593536f78d3ddaf4cf712101d3fb6d981c6cc55555ad686a9adbe3397a727f62f561e8d8755bdcd88a763777da30281cc2924fc160b0386c3f99f5bd9",
