diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9660f55..c3edd53 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,6 +11,8 @@ project(opengv VERSION 1.0 LANGUAGES CXX)
 
 set(CMAKE_BUILD_TYPE Release)
 
+include(GNUInstallDirs)
+
 #set the default path for built executables to the "bin" directory
 set(EXECUTABLE_OUTPUT_PATH ${CMAKE_BINARY_DIR}/bin)
 #set the default path for built libraries to the "lib" directory
@@ -337,8 +339,8 @@ IF (BUILD_TESTS)
 ENDIF()
 
 # Configuration
-set(config_install_dir "lib/cmake/${PROJECT_NAME}-${PROJECT_VERSION}")
-set(include_install_dir "include")
+set(config_install_dir "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
+set(include_install_dir "${CMAKE_INSTALL_INCLUDEDIR}")
 set(version_config "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake")
 set(project_config "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake")
 set(targets_export_name "${PROJECT_NAME}Targets")
@@ -366,10 +368,10 @@ export(TARGETS opengv FILE "${CMAKE_CURRENT_BINARY_DIR}/${targets_export_name}.c
 # Targets:
 install(TARGETS opengv
         EXPORT "${targets_export_name}"
-        LIBRARY DESTINATION "lib"
-        ARCHIVE DESTINATION "lib"
-        RUNTIME DESTINATION "bin"
-        INCLUDES DESTINATION "${include_install_dir}")
+        LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+        ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+        RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+        INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
 
 # Config
 #   * <prefix>/lib/cmake/opengv/opengvConfig.cmake
@@ -385,7 +387,7 @@ install(EXPORT "${targets_export_name}"
 
 # Headers
 install(DIRECTORY include/ 
-        DESTINATION ${include_install_dir} 
+        DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
         FILES_MATCHING PATTERN "*.h" PATTERN "*.hpp")
 
 
