diff --git a/examples/export-lora/export-lora.cpp b/examples/export-lora/export-lora.cpp
index 8df457e2..d27e942c 100644
--- a/examples/export-lora/export-lora.cpp
+++ b/examples/export-lora/export-lora.cpp
@@ -132,7 +132,7 @@ struct lora_merge_ctx {
             int n_threads) : base_model(base_fname, 0), n_threads(n_threads), fout(outfile, std::ios::binary) {
         fout.exceptions(std::ofstream::failbit); // fail fast on write errors
 
-        if (gguf_find_key(base_model.ctx_gguf, LLM_KV_SPLIT_COUNT) >= 0) {
+        if (gguf_find_key(base_model.ctx_gguf, LLM_KV_SPLIT_COUNT_STR) >= 0) {
             throw std::runtime_error("split model is not yet supported");
         }
 
