diff --git a/src/external/CMakeLists.txt b/src/external/CMakeLists.txt
index 0705627ae..a2b1eadc5 100644
--- a/src/external/CMakeLists.txt
+++ b/src/external/CMakeLists.txt
@@ -198,12 +198,12 @@ target_include_directories(
 
 # Tracy
 if(XRT_HAVE_TRACY)
-	add_library(xrt-external-tracy STATIC tracy/TracyClient.cpp)
-	target_include_directories(
-		xrt-external-tracy SYSTEM PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/tracy
-		)
-	target_compile_definitions(xrt-external-tracy PRIVATE TRACY_ENABLE)
-	target_link_libraries(xrt-external-tracy PRIVATE ${CMAKE_DL_LIBS})
+	pkg_search_module(TRACY REQUIRED tracy)
+	add_library(xrt-external-tracy INTERFACE)
+	target_include_directories(xrt-external-tracy INTERFACE ${TRACY_INCLUDEDIR})
+	target_compile_options(xrt-external-tracy INTERFACE ${TRACY_CFLAGS})
+	target_link_options(xrt-external-tracy INTERFACE ${TRACY_LDFLAGS})
+	target_link_libraries(xrt-external-tracy INTERFACE ${TRACY_LIBRARIES} ${TRACY_LINK_LIBRARIES})
 endif()
 
 # Valve VDF parser
