diff --git a/meson.build b/meson.build
index a0b1bd5..a7c5d50 100644
--- a/meson.build
+++ b/meson.build
@@ -204,9 +204,9 @@ tracy = library('tracy', tracy_src, tracy_header_files,
     override_options    : override_options,
     install             : true)
 
-install_headers(includes, subdir : 'tracy')
-install_headers(common_includes, subdir : 'common')
-install_headers(client_includes, subdir : 'client')
+install_headers(includes, subdir : 'tracy/tracy')
+install_headers(common_includes, subdir : 'tracy/common')
+install_headers(client_includes, subdir : 'tracy/client')
 
 tracy_dep_compile_args = tracy_common_args
 
@@ -217,6 +217,7 @@ endif
 pkg = import('pkgconfig')
 pkg.generate(tracy,
   extra_cflags : tracy_dep_compile_args,
+  subdirs      : 'tracy',
   requires     : tracy_public_deps)
 
 tracy_dep = declare_dependency(
