diff --git a/meson.build b/meson.build
index fa57b4b..fbf8192 100644
--- a/meson.build
+++ b/meson.build
@@ -147,6 +147,7 @@ client_includes = [
     'public/client/TracyArmCpuTable.hpp',
     'public/client/TracyCallstack.h',
     'public/client/TracyCallstack.hpp',
+    'public/client/TracyCpuid.hpp',
     'public/client/TracyDebug.hpp',
     'public/client/TracyDxt1.hpp',
     'public/client/TracyFastVector.hpp',
@@ -177,6 +178,7 @@ common_includes = [
     'public/common/TracyStackFrames.hpp',
     'public/common/TracySystem.hpp',
     'public/common/TracyUwp.hpp',
+    'public/common/TracyVersion.hpp',
     'public/common/TracyYield.hpp'
 ]
 
