diff --git a/SConstruct b/SConstruct
index d0ffcde3e3..4ca0926dee 100644
--- a/SConstruct
+++ b/SConstruct
@@ -196,6 +196,7 @@ opts.Add(BoolVariable("builtin_zstd", "Use the built-in Zstd library", True))
 # Compilation environment setup
 opts.Add("CXX", "C++ compiler")
 opts.Add("CC", "C compiler")
+opts.Add("AR", "Archiver")
 opts.Add("LINK", "Linker")
 opts.Add("CCFLAGS", "Custom flags for both the C and C++ compilers")
 opts.Add("CFLAGS", "Custom flags for the C compiler")
diff --git a/platform/server/detect.py b/platform/server/detect.py
index 7bd5a0fd77..2e90b71d7b 100644
--- a/platform/server/detect.py
+++ b/platform/server/detect.py
@@ -220,6 +220,9 @@ def configure(env):
     if not env["builtin_enet"]:
         env.ParseConfig("pkg-config libenet --cflags --libs")
 
+    if not env["builtin_recast"]:
+        env.ParseConfig("pkg-config recastnavigation --cflags --libs")
+
     if not env["builtin_squish"]:
         env.ParseConfig("pkg-config libsquish --cflags --libs")
 
@@ -247,7 +250,7 @@ def configure(env):
 
     if not env["builtin_opus"]:
         env["builtin_libogg"] = False  # Needed to link against system opus
-        env.ParseConfig("pkg-config opus opusfile --cflags --libs")
+        env.ParseConfig("pkg-config opus --cflags --libs")
 
     if not env["builtin_libogg"]:
         env.ParseConfig("pkg-config ogg --cflags --libs")
diff --git a/platform/x11/detect.py b/platform/x11/detect.py
index 6390c263a7..98fa7a0e09 100644
--- a/platform/x11/detect.py
+++ b/platform/x11/detect.py
@@ -241,18 +241,6 @@ def configure(env):
 
     env.Append(CCFLAGS=["-pipe"])
 
-    # Check for gcc version >= 6 before adding -no-pie
-    version = get_compiler_version(env) or [-1, -1]
-    if using_gcc(env):
-        if version[0] >= 6:
-            env.Append(CCFLAGS=["-fpie"])
-            env.Append(LINKFLAGS=["-no-pie"])
-    # Do the same for clang should be fine with Clang 4 and higher
-    if using_clang(env):
-        if version[0] >= 4:
-            env.Append(CCFLAGS=["-fpie"])
-            env.Append(LINKFLAGS=["-no-pie"])
-
     ## Dependencies
 
     env.ParseConfig("pkg-config x11 --cflags --libs")
@@ -305,6 +293,9 @@ def configure(env):
     if not env["builtin_enet"]:
         env.ParseConfig("pkg-config libenet --cflags --libs")
 
+    if not env["builtin_recast"]:
+        env.ParseConfig("pkg-config recastnavigation --cflags --libs")
+
     if not env["builtin_squish"]:
         env.ParseConfig("pkg-config libsquish --cflags --libs")
 
@@ -332,7 +323,7 @@ def configure(env):
 
     if not env["builtin_opus"]:
         env["builtin_libogg"] = False  # Needed to link against system opus
-        env.ParseConfig("pkg-config opus opusfile --cflags --libs")
+        env.ParseConfig("pkg-config opus --cflags --libs")
 
     if not env["builtin_libogg"]:
         env.ParseConfig("pkg-config ogg --cflags --libs")
