diff --git a/platform/linuxbsd/wayland/display_server_wayland.cpp b/platform/linuxbsd/wayland/display_server_wayland.cpp
index f08c0fcac4..11b57ed2e2 100644
--- a/platform/linuxbsd/wayland/display_server_wayland.cpp
+++ b/platform/linuxbsd/wayland/display_server_wayland.cpp
@@ -87,6 +87,10 @@
 
 #define WAYLAND_MAX_FRAME_TIME_US (1'000'000)
 
+#ifdef CursorShape
+#undef CursorShape
+#endif
+
 String DisplayServerWayland::_get_app_id_from_context(Context p_context) {
 	String app_id;
 
