diff --git a/modules/mono/godotsharp_dirs.cpp b/modules/mono/godotsharp_dirs.cpp
index 039263b405..002089fd88 100644
--- a/modules/mono/godotsharp_dirs.cpp
+++ b/modules/mono/godotsharp_dirs.cpp
@@ -147,7 +147,7 @@ private:
 		mono_user_dir = _get_mono_user_dir();
 #endif
 
-		String exe_dir = OS::get_singleton()->get_executable_path().get_base_dir();
+		String exe_dir = "/usr/share/godot/__GODOT_VERSION__/";
 		String res_dir = OS::get_singleton()->get_bundle_resource_dir();
 
 #ifdef TOOLS_ENABLED
