diff --git a/otherarch/sdcpp/sdtype_adapter.cpp b/otherarch/sdcpp/sdtype_adapter.cpp
index 237ed93e..c4a3c538 100644
--- a/otherarch/sdcpp/sdtype_adapter.cpp
+++ b/otherarch/sdcpp/sdtype_adapter.cpp
@@ -157,7 +157,7 @@ bool sdtype_load_model(const sd_load_model_inputs inputs) {
     printf("\nImageGen Init - Load Model: %s\n",inputs.model_filename);
     if(lorafilename!="")
     {
-        printf("With LoRA: %s at %f power\n",lorafilename.c_str(),inputs.lora_multiplier);
+        printf("With LoRA directory: %s\n",lorafilename.c_str());
     }
     if(inputs.taesd)
     {
@@ -202,6 +202,10 @@ bool sdtype_load_model(const sd_load_model_inputs inputs) {
     sd_params->batch_count = 1;
     sd_params->vae_path = vaefilename;
     sd_params->taesd_path = taesdpath;
+    if(lorafilename!="")
+    {
+        sd_params->lora_model_dir = lorafilename;
+    }
 
     sddebugmode = inputs.debugmode;
 
@@ -248,13 +252,6 @@ bool sdtype_load_model(const sd_load_model_inputs inputs) {
         return false;
     }
 
-    if(lorafilename!="" && inputs.lora_multiplier>0)
-    {
-        printf("\nApply LoRA...\n");
-       // sd_ctx->sd->set_pending_lora(lorafilename,inputs.lora_multiplier);
-        sd_ctx->sd->apply_lora_from_file(lorafilename,inputs.lora_multiplier);
-    }
-
     return true;
 
 }
