diff --git a/src/config/defaults.ts b/src/config/defaults.ts
index dcd5612..27709da 100644
--- a/src/config/defaults.ts
+++ b/src/config/defaults.ts
@@ -52,7 +52,7 @@ const defaultConfig: DefaultConfig = {
   'options': {
     tray: false,
     appVisible: true,
-    autoUpdates: true,
+    autoUpdates: false,
     alwaysOnTop: false,
     hideMenu: false,
     hideMenuWarned: false,
diff --git a/src/index.ts b/src/index.ts
index afc4e2f..ba0986e 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -766,57 +766,6 @@ app.whenReady().then(async () => {
     openAtLogin: config.get('options.startAtLogin'),
   });
 
-  if (!is.dev() && config.get('options.autoUpdates')) {
-    const updateTimeout = setTimeout(() => {
-      autoUpdater.checkForUpdatesAndNotify();
-      clearTimeout(updateTimeout);
-    }, 2000);
-    autoUpdater.on('update-available', () => {
-      const downloadLink =
-        'https://github.com/th-ch/youtube-music/releases/latest';
-      const dialogOptions: Electron.MessageBoxOptions = {
-        type: 'info',
-        buttons: [
-          t('main.dialog.update-available.buttons.ok'),
-          t('main.dialog.update-available.buttons.download'),
-          t('main.dialog.update-available.buttons.disable'),
-        ],
-        title: t('main.dialog.update-available.title'),
-        message: t('main.dialog.update-available.message'),
-        detail: t('main.dialog.update-available.detail', { downloadLink }),
-        defaultId: 1,
-        cancelId: 0,
-      };
-
-      let dialogPromise: Promise<Electron.MessageBoxReturnValue>;
-      if (mainWindow) {
-        dialogPromise = dialog.showMessageBox(mainWindow, dialogOptions);
-      } else {
-        dialogPromise = dialog.showMessageBox(dialogOptions);
-      }
-
-      dialogPromise.then((dialogOutput) => {
-        switch (dialogOutput.response) {
-          // Download
-          case 1: {
-            shell.openExternal(downloadLink);
-            break;
-          }
-
-          // Disable updates
-          case 2: {
-            config.set('options.autoUpdates', false);
-            break;
-          }
-
-          case 0: {
-            break;
-          }
-        }
-      });
-    });
-  }
-
   if (config.get('options.hideMenu') && !config.get('options.hideMenuWarned')) {
     dialog.showMessageBox(mainWindow, {
       type: 'info',
diff --git a/src/menu.ts b/src/menu.ts
index 61fd546..943d27a 100644
--- a/src/menu.ts
+++ b/src/menu.ts
@@ -156,14 +156,6 @@ export const mainMenuTemplate = async (
     {
       label: t('main.menu.options.label'),
       submenu: [
-        {
-          label: t('main.menu.options.submenu.auto-update'),
-          type: 'checkbox',
-          checked: config.get('options.autoUpdates'),
-          click(item: MenuItem) {
-            config.setMenuOption('options.autoUpdates', item.checked);
-          },
-        },
         {
           label: t('main.menu.options.submenu.resume-on-start'),
           type: 'checkbox',
