diff --git a/examples/bench/bench.cpp b/examples/bench/bench.cpp
index 949e573..f303d55 100644
--- a/examples/bench/bench.cpp
+++ b/examples/bench/bench.cpp
@@ -10,7 +10,7 @@ struct whisper_params {
     int32_t n_threads = std::min(4, (int32_t) std::thread::hardware_concurrency());
     int32_t what = 0; // what to benchmark: 0 - whisper ecoder, 1 - memcpy, 2 - ggml_mul_mat
 
-    std::string model = "models/ggml-base.en.bin";
+    std::string model = "/usr/share/whisper/ggml-models/base.en.bin";
 
     bool use_gpu = true;
 };
diff --git a/examples/command/command.cpp b/examples/command/command.cpp
index 51d800a..60b3cb8 100644
--- a/examples/command/command.cpp
+++ b/examples/command/command.cpp
@@ -51,7 +51,7 @@ struct whisper_params {
     bool use_gpu       = true;
 
     std::string language  = "en";
-    std::string model     = "models/ggml-base.en.bin";
+    std::string model     = "/usr/share/whisper/ggml-models/base.en.bin";
     std::string fname_out;
     std::string commands;
     std::string prompt;
diff --git a/examples/lsp/lsp.cpp b/examples/lsp/lsp.cpp
index 8d8b6ff..1e4e43e 100644
--- a/examples/lsp/lsp.cpp
+++ b/examples/lsp/lsp.cpp
@@ -33,7 +33,7 @@ struct whisper_params {
     bool use_gpu       = true;
 
     std::string language  = "en";
-    std::string model     = "models/ggml-base.en.bin";
+    std::string model     = "/usr/share/whisper/ggml-models/base.en.bin";
 };
 struct command {
     std::vector<whisper_token> tokens;
diff --git a/examples/lsp/whisper.vim b/examples/lsp/whisper.vim
index 07ee0aa..9eda2e9 100644
--- a/examples/lsp/whisper.vim
+++ b/examples/lsp/whisper.vim
@@ -13,7 +13,7 @@ if !exists("g:whisper_lsp_path")
 endif
 if !exists("g:whisper_model_path")
     " TODO: allow custom paths relative to the repo dir
-    let g:whisper_model_path = g:whisper_dir .. "models/ggml-base.en.bin"
+    let g:whisper_model_path = g:whisper_dir .. "/usr/share/whisper/ggml-models/base.en.bin"
     if !filereadable(g:whisper_model_path)
         echoerr "Could not find model at: " .. g:whisper_model_path
         throw "Model not found"
diff --git a/examples/main/main.cpp b/examples/main/main.cpp
index 9699802..e29a637 100644
--- a/examples/main/main.cpp
+++ b/examples/main/main.cpp
@@ -95,7 +95,7 @@ struct whisper_params {
     std::string language  = "en";
     std::string prompt;
     std::string font_path = "/System/Library/Fonts/Supplemental/Courier New Bold.ttf";
-    std::string model     = "models/ggml-base.en.bin";
+    std::string model     = "/usr/share/whisper/ggml-models/base.en.bin";
 
     // [TDRZ] speaker turn string
     std::string tdrz_speaker_turn = " [SPEAKER_TURN]"; // TODO: set from command line
diff --git a/examples/server/server.cpp b/examples/server/server.cpp
index ac2f2a5..3702ec8 100644
--- a/examples/server/server.cpp
+++ b/examples/server/server.cpp
@@ -83,7 +83,7 @@ struct whisper_params {
     std::string language        = "en";
     std::string prompt          = "";
     std::string font_path       = "/System/Library/Fonts/Supplemental/Courier New Bold.ttf";
-    std::string model           = "models/ggml-base.en.bin";
+    std::string model           = "/usr/share/whisper/ggml-models/base.en.bin";
 
     std::string response_format     = json_format;
 
diff --git a/examples/stream/stream.cpp b/examples/stream/stream.cpp
index 47f1780..0638726 100644
--- a/examples/stream/stream.cpp
+++ b/examples/stream/stream.cpp
@@ -52,7 +52,7 @@ struct whisper_params {
     bool use_gpu       = true;
 
     std::string language  = "en";
-    std::string model     = "models/ggml-base.en.bin";
+    std::string model     = "/usr/share/whisper/ggml-models/base.en.bin";
     std::string fname_out;
 };
 
diff --git a/examples/talk-llama/talk-llama.cpp b/examples/talk-llama/talk-llama.cpp
index 07d9855..be62500 100644
--- a/examples/talk-llama/talk-llama.cpp
+++ b/examples/talk-llama/talk-llama.cpp
@@ -68,7 +68,7 @@ struct whisper_params {
 
     std::string person      = "Georgi";
     std::string language    = "en";
-    std::string model_wsp   = "models/ggml-base.en.bin";
+    std::string model_wsp   = "/usr/share/whisper/ggml-models/base.en.bin";
     std::string model_llama = "models/ggml-llama-7B.bin";
     std::string speak       = "./examples/talk-llama/speak";
     std::string prompt      = "";
diff --git a/examples/talk/talk.cpp b/examples/talk/talk.cpp
index cdb1a23..9946b7f 100644
--- a/examples/talk/talk.cpp
+++ b/examples/talk/talk.cpp
@@ -35,7 +35,7 @@ struct whisper_params {
 
     std::string person    = "Santa";
     std::string language  = "en";
-    std::string model_wsp = "models/ggml-base.en.bin";
+    std::string model_wsp = "/usr/share/whisper/ggml-models/base.en.bin";
     std::string model_gpt = "models/ggml-gpt-2-117M.bin";
     std::string speak     = "./examples/talk/speak";
     std::string fname_out;
diff --git a/examples/wchess/wchess.cmd/wchess.cmd.cpp b/examples/wchess/wchess.cmd/wchess.cmd.cpp
index 88771b4..b3b1bea 100644
--- a/examples/wchess/wchess.cmd/wchess.cmd.cpp
+++ b/examples/wchess/wchess.cmd/wchess.cmd.cpp
@@ -34,7 +34,7 @@ struct whisper_params {
     bool use_gpu       = true;
 
     std::string language  = "en";
-    std::string model     = "models/ggml-base.en.bin";
+    std::string model     = "/usr/share/whisper/ggml-models/base.en.bin";
     std::string fname_out;
     std::string commands;
     std::string prompt;
