diff --git a/CMakeLists.txt b/CMakeLists.txt
index 567a58d..cf0f947 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -184,7 +184,7 @@ if (WHISPER_BLAS)
             message(FATAL_ERROR "BLAS library was not found. Environment variable OPENBLAS_PATH not defined.")
         endif ()
     else ()
-        set(BLA_STATIC 1)
+        set(BLA_STATIC OFF)
         set(BLA_VENDOR ${WHISPER_BLAS_VENDOR})
         set(BLA_SIZEOF_INTEGER 8)
         set(BLA_PREFER_PKGCONFIG 1)
