diff --git a/meson.build b/meson.build
index 18cef1d..ce85696 100644
--- a/meson.build
+++ b/meson.build
@@ -179,23 +179,24 @@ endif
 
 # Commented code can be used if mangohud start using latest SDK Vulkan-Headers
 # Allowing user to build mangohud using system Vulkan-Headers
-# if not dependency('VulkanHeaders').found()
-vkh_sp = subproject('vulkan-headers')
-vk_api_xml = vkh_sp.get_variable('vulkan_api_xml')
-vulkan_headers_dep = vkh_sp.get_variable('vulkan_headers_dep')
 vulkan_lib = cc.find_library('vulkan', required: true)
+if not dependency('VulkanHeaders').found()
+  vkh_sp = subproject('vulkan-headers')
+  vk_api_xml = vkh_sp.get_variable('vulkan_api_xml')
+  vulkan_headers_dep = vkh_sp.get_variable('vulkan_headers_dep')
+  vulkan_utility_libraries_sp = subproject('vulkan-utility-libraries')
+  dep_vulkan_utility_libraries = vulkan_utility_libraries_sp.get_variable('vulkan_utility_libraries_dep')
+else
+  vk_api_xml = files('/usr/share/vulkan/registry/vk.xml')
+  vulkan_headers_dep = dependency('VulkanHeaders', required: true)
+  dep_vulkan_utility_libraries = dependency('VulkanUtilityLibraries', required: false)
+endif
+
 vulkan_dep = declare_dependency(
   dependencies: [vulkan_headers_dep, vulkan_lib]
 )
 meson.override_dependency('vulkan', vulkan_dep)
 
-  vulkan_utility_libraries_sp = subproject('vulkan-utility-libraries')
-  dep_vulkan_utility_libraries = vulkan_utility_libraries_sp.get_variable('vulkan_utility_libraries_dep')
-# else
-#  dep_vulkan_headers = dependency('VulkanHeaders', required: true)
-#  vulkan_headers_api_xml = files('/usr/share/vulkan/registry/vk.xml')
-# endif
-
 vk_enum_to_str = custom_target(
   'vk_enum_to_str',
   input : ['bin/gen_enum_to_str.py', vk_api_xml],
