diff --git a/src/hud_elements.cpp b/src/hud_elements.cpp
index 0c89987..191ecb7 100644
--- a/src/hud_elements.cpp
+++ b/src/hud_elements.cpp
@@ -406,7 +406,7 @@ void HudElements::core_load(){
             width = 150;
             height = HUDElements.params->font_size;
         } else {
-            width = ImGui::GetWindowContentRegionWidth();
+            width = ImGui::GetContentRegionAvail().x;
             height = 50;
         }
 
@@ -793,7 +793,7 @@ void HudElements::frame_timing(){
             width = 150;
             height = HUDElements.params->font_size * 0.85;
         } else {
-            width = ImGui::GetWindowContentRegionWidth();
+            width = ImGui::GetContentRegionAvail().x;
             height = max_time;
         }
 
@@ -1133,11 +1133,11 @@ void HudElements::gamescope_frame_timing(){
             HUDElements.sw_stats->time_dividor = 1000000.0f; /* ns -> ms */
 
             ImGui::PushStyleColor(ImGuiCol_FrameBg, ImVec4(0.0f, 0.0f, 0.0f, 0.0f));
-            if (ImGui::BeginChild("gamescope_app_window", ImVec2(ImGui::GetWindowContentRegionWidth(), 50))) {
+            if (ImGui::BeginChild("gamescope_app_window", ImVec2(ImGui::GetContentRegionAvail().x, 50))) {
                 ImGui::PlotLines("", HUDElements.gamescope_debug_app.data(),
                         HUDElements.gamescope_debug_app.size(), 0,
                         NULL, min_time, max_time,
-                            ImVec2(ImGui::GetWindowContentRegionWidth(), 50));
+                            ImVec2(ImGui::GetContentRegionAvail().x, 50));
             }
             ImGui::PopStyleColor();
             ImGui::EndChild();
@@ -1158,11 +1158,11 @@ void HudElements::gamescope_frame_timing(){
             ImguiNextColumnFirstItem();
             ImGui::PushStyleColor(ImGuiCol_FrameBg, ImVec4(0.0f, 0.0f, 0.0f, 0.0f));
             ImGui::PushStyleColor(ImGuiCol_PlotLines, ImVec4(0,0,1,1));
-            if (ImGui::BeginChild("gamescope_latency_window", ImVec2(ImGui::GetWindowContentRegionWidth(), 50))) {
+            if (ImGui::BeginChild("gamescope_latency_window", ImVec2(ImGui::GetContentRegionAvail().x, 50))) {
                 ImGui::PlotLines("", HUDElements.gamescope_debug_latency.data(),
                         HUDElements.gamescope_debug_latency.size(), 0,
                         NULL, min_time, max_time,
-                        ImVec2(ImGui::GetWindowContentRegionWidth(), 50));
+                        ImVec2(ImGui::GetContentRegionAvail().x, 50));
             }
             ImGui::PopStyleColor(2);
             ImGui::EndChild();
@@ -1393,12 +1393,12 @@ void HudElements::graphs(){
         ImGui::PlotLines("", arr.data(),
                 arr.size(), 0,
                 NULL, HUDElements.min, HUDElements.max,
-                ImVec2(ImGui::GetWindowContentRegionWidth(), 50));
+                ImVec2(ImGui::GetContentRegionAvail().x, 50));
     } else {
         ImGui::PlotHistogram("", arr.data(),
             arr.size(), 0,
             NULL, HUDElements.min, HUDElements.max,
-            ImVec2(ImGui::GetWindowContentRegionWidth(), 50));
+            ImVec2(ImGui::GetContentRegionAvail().x, 50));
     }
     ImGui::Dummy(ImVec2(0.0f,5.0f));
     ImGui::PopStyleColor(1);
