diff --git a/cmake/build_helpers.cmake b/cmake/build_helpers.cmake
index 384fe96e0..3391f9215 100644
--- a/cmake/build_helpers.cmake
+++ b/cmake/build_helpers.cmake
@@ -665,6 +665,10 @@ function(downloadImHexPatternsFiles dest)
         if (NOT EXISTS ${imhex_patterns_SOURCE_DIR})
             set(imhex_patterns_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../ImHex-Patterns")
         endif()
+
+        if (NOT EXISTS ${imhex_patterns_SOURCE_DIR})
+            set(imhex_patterns_SOURCE_DIR "")
+        endif()
     endif ()
 
     install(CODE "set(imhex_patterns_SOURCE_DIR \"${imhex_patterns_SOURCE_DIR}\")")
