diff --git a/lib/rlib/rchunk.hpp b/lib/rlib/rchunk.hpp
index f42890f..0c84915 100644
--- a/lib/rlib/rchunk.hpp
+++ b/lib/rlib/rchunk.hpp
@@ -67,7 +67,7 @@ namespace rlib {
 template <>
 struct fmt::formatter<rlib::BundleID> : formatter<std::string> {
     template <typename FormatContext>
-    auto format(rlib::BundleID id, FormatContext& ctx) {
+    auto format(rlib::BundleID id, FormatContext& ctx) const {
         return formatter<std::string>::format(fmt::format("{:016X}", (std::uint64_t)id), ctx);
     }
 };
@@ -75,7 +75,7 @@ struct fmt::formatter<rlib::BundleID> : formatter<std::string> {
 template <>
 struct fmt::formatter<rlib::ChunkID> : formatter<std::string> {
     template <typename FormatContext>
-    auto format(rlib::ChunkID id, FormatContext& ctx) {
+    auto format(rlib::ChunkID id, FormatContext& ctx) const {
         return formatter<std::string>::format(fmt::format("{:016X}", (std::uint64_t)id), ctx);
     }
 };
diff --git a/lib/rlib/rfile.hpp b/lib/rlib/rfile.hpp
index a2f1ebe..192010b 100644
--- a/lib/rlib/rfile.hpp
+++ b/lib/rlib/rfile.hpp
@@ -53,7 +53,7 @@ namespace rlib {
 template <>
 struct fmt::formatter<rlib::FileID> : formatter<std::string> {
     template <typename FormatContext>
-    auto format(rlib::FileID id, FormatContext& ctx) {
+    auto format(rlib::FileID id, FormatContext& ctx) const {
         return formatter<std::string>::format(fmt::format("{:016X}", (std::uint64_t)id), ctx);
     }
 };
