diff --git a/lib/rlib/rchunk.cpp b/lib/rlib/rchunk.cpp
index 665908a..e5f9d14 100644
--- a/lib/rlib/rchunk.cpp
+++ b/lib/rlib/rchunk.cpp
@@ -2,7 +2,7 @@
 
 #include <bit>
 #include <cstring>
-#include <digestpp.hpp>
+#include <digestpp/digestpp.hpp>
 
 #include "common.hpp"
 #include "iofile.hpp"
@@ -104,4 +104,4 @@ auto RChunk::Dst::verify(fs::path const& path, std::vector<RChunk::Dst>& chunks,
         }
         return false;
     });
-}
\ No newline at end of file
+}
