diff --git a/lib/rlib/rbundle.cpp b/lib/rlib/rbundle.cpp
index 4acb18b..15e286b 100644
--- a/lib/rlib/rbundle.cpp
+++ b/lib/rlib/rbundle.cpp
@@ -1,6 +1,6 @@
 #include "rbundle.hpp"
 
-#include <common/xxhash.h>
+#include <xxhash.h>
 #include <zstd.h>
 
 #include <bit>
diff --git a/lib/rlib/rcache.cpp b/lib/rlib/rcache.cpp
index 0fcb87f..84b0f31 100644
--- a/lib/rlib/rcache.cpp
+++ b/lib/rlib/rcache.cpp
@@ -1,6 +1,6 @@
 #include "rcache.hpp"
 
-#include <common/xxhash.h>
+#include <xxhash.h>
 #include <zstd.h>
 
 #include <charconv>
@@ -303,4 +303,4 @@ auto RCache::load_folder_internal() -> void {
         }
         lookup_.merge(std::move(bundle.lookup));
     }
-}
\ No newline at end of file
+}
diff --git a/src/rman_make.cpp b/src/rman_make.cpp
index 72a7f61..d362574 100644
--- a/src/rman_make.cpp
+++ b/src/rman_make.cpp
@@ -1,4 +1,4 @@
-#include <common/xxhash.h>
+#include <xxhash.h>
 #include <fmt/args.h>
 #include <fmt/format.h>
 
diff --git a/src/rman_merge.cpp b/src/rman_merge.cpp
index 8e49198..ba910a0 100644
--- a/src/rman_merge.cpp
+++ b/src/rman_merge.cpp
@@ -1,4 +1,4 @@
-#include <common/xxhash.h>
+#include <xxhash.h>
 #include <fmt/args.h>
 #include <fmt/format.h>
 
diff --git a/src/rman_remake.cpp b/src/rman_remake.cpp
index f63b39e..a4bbf58 100644
--- a/src/rman_remake.cpp
+++ b/src/rman_remake.cpp
@@ -1,4 +1,4 @@
-#include <common/xxhash.h>
+#include <xxhash.h>
 #include <fmt/args.h>
 #include <fmt/format.h>
 
