diff --git a/src/rbun_chk.cpp b/src/rbun_chk.cpp
index 547bac0..3b647c4 100644
--- a/src/rbun_chk.cpp
+++ b/src/rbun_chk.cpp
@@ -1,4 +1,4 @@
-#include <argparse.hpp>
+#include <argparse/argparse.hpp>
 #include <iostream>
 #include <rlib/common.hpp>
 #include <rlib/iofile.hpp>
diff --git a/src/rbun_ex.cpp b/src/rbun_ex.cpp
index f7b3693..423e8ea 100644
--- a/src/rbun_ex.cpp
+++ b/src/rbun_ex.cpp
@@ -1,4 +1,4 @@
-#include <argparse.hpp>
+#include <argparse/argparse.hpp>
 #include <charconv>
 #include <iostream>
 #include <rlib/common.hpp>
diff --git a/src/rbun_ls.cpp b/src/rbun_ls.cpp
index 9c64517..4b6b8a5 100644
--- a/src/rbun_ls.cpp
+++ b/src/rbun_ls.cpp
@@ -1,4 +1,4 @@
-#include <argparse.hpp>
+#include <argparse/argparse.hpp>
 #include <iostream>
 #include <rlib/common.hpp>
 #include <rlib/iofile.hpp>
diff --git a/src/rbun_merge.cpp b/src/rbun_merge.cpp
index 22da14d..ecc328d 100644
--- a/src/rbun_merge.cpp
+++ b/src/rbun_merge.cpp
@@ -1,4 +1,4 @@
-#include <argparse.hpp>
+#include <argparse/argparse.hpp>
 #include <iostream>
 #include <rlib/common.hpp>
 #include <rlib/iofile.hpp>
diff --git a/src/rbun_usage.cpp b/src/rbun_usage.cpp
index cfd943b..25558ad 100644
--- a/src/rbun_usage.cpp
+++ b/src/rbun_usage.cpp
@@ -1,4 +1,4 @@
-#include <argparse.hpp>
+#include <argparse/argparse.hpp>
 #include <iostream>
 #include <rlib/common.hpp>
 #include <rlib/iofile.hpp>
diff --git a/src/rman_bl.cpp b/src/rman_bl.cpp
index a2bd612..8349883 100644
--- a/src/rman_bl.cpp
+++ b/src/rman_bl.cpp
@@ -1,7 +1,7 @@
 #include <fmt/args.h>
 #include <fmt/format.h>
 
-#include <argparse.hpp>
+#include <argparse/argparse.hpp>
 #include <iostream>
 #include <rlib/common.hpp>
 #include <rlib/iofile.hpp>
diff --git a/src/rman_chk.cpp b/src/rman_chk.cpp
index b8af7ca..198b2aa 100644
--- a/src/rman_chk.cpp
+++ b/src/rman_chk.cpp
@@ -1,7 +1,7 @@
 #include <fmt/args.h>
 #include <fmt/format.h>
 
-#include <argparse.hpp>
+#include <argparse/argparse.hpp>
 #include <iostream>
 #include <rlib/common.hpp>
 #include <rlib/iofile.hpp>
diff --git a/src/rman_diff.cpp b/src/rman_diff.cpp
index 81263a8..3dd3dad 100644
--- a/src/rman_diff.cpp
+++ b/src/rman_diff.cpp
@@ -1,4 +1,4 @@
-#include <argparse.hpp>
+#include <argparse/argparse.hpp>
 #include <iostream>
 #include <rlib/common.hpp>
 #include <rlib/iofile.hpp>
diff --git a/src/rman_dl.cpp b/src/rman_dl.cpp
index 6c91be8..d187ca1 100644
--- a/src/rman_dl.cpp
+++ b/src/rman_dl.cpp
@@ -1,4 +1,4 @@
-#include <argparse.hpp>
+#include <argparse/argparse.hpp>
 #include <iostream>
 #include <rlib/common.hpp>
 #include <rlib/iofile.hpp>
diff --git a/src/rman_ls.cpp b/src/rman_ls.cpp
index 58b63a6..a029f0c 100644
--- a/src/rman_ls.cpp
+++ b/src/rman_ls.cpp
@@ -1,4 +1,4 @@
-#include <argparse.hpp>
+#include <argparse/argparse.hpp>
 #include <iostream>
 #include <rlib/common.hpp>
 #include <rlib/iofile.hpp>
diff --git a/src/rman_make.cpp b/src/rman_make.cpp
index 918b20d..72a7f61 100644
--- a/src/rman_make.cpp
+++ b/src/rman_make.cpp
@@ -2,7 +2,7 @@
 #include <fmt/args.h>
 #include <fmt/format.h>
 
-#include <argparse.hpp>
+#include <argparse/argparse.hpp>
 #include <iostream>
 #include <rlib/ar.hpp>
 #include <rlib/common.hpp>
diff --git a/src/rman_merge.cpp b/src/rman_merge.cpp
index 997ac21..8e49198 100644
--- a/src/rman_merge.cpp
+++ b/src/rman_merge.cpp
@@ -2,7 +2,7 @@
 #include <fmt/args.h>
 #include <fmt/format.h>
 
-#include <argparse.hpp>
+#include <argparse/argparse.hpp>
 #include <iostream>
 #include <rlib/ar.hpp>
 #include <rlib/common.hpp>
diff --git a/src/rman_mount.cpp b/src/rman_mount.cpp
index d2c0c2a..0c0f19c 100644
--- a/src/rman_mount.cpp
+++ b/src/rman_mount.cpp
@@ -4,7 +4,7 @@
 #include <fuse3/fuse_common.h>
 #include <sys/stat.h>
 
-#include <argparse.hpp>
+#include <argparse/argparse.hpp>
 #include <chrono>
 #include <cinttypes>
 #include <compare>
diff --git a/src/rman_rads.cpp b/src/rman_rads.cpp
index e84d89e..c4966c7 100644
--- a/src/rman_rads.cpp
+++ b/src/rman_rads.cpp
@@ -1,7 +1,7 @@
 #include <fmt/args.h>
 #include <fmt/format.h>
 
-#include <argparse.hpp>
+#include <argparse/argparse.hpp>
 #include <iostream>
 #include <rlib/common.hpp>
 #include <rlib/iofile.hpp>
diff --git a/src/rman_remake.cpp b/src/rman_remake.cpp
index 12f76c6..f63b39e 100644
--- a/src/rman_remake.cpp
+++ b/src/rman_remake.cpp
@@ -2,7 +2,7 @@
 #include <fmt/args.h>
 #include <fmt/format.h>
 
-#include <argparse.hpp>
+#include <argparse/argparse.hpp>
 #include <iostream>
 #include <rlib/ar.hpp>
 #include <rlib/common.hpp>
