diff --git a/CMakeLists.txt b/CMakeLists.txt
index ff7717c7..0326fe1c 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -858,6 +858,8 @@ add_definitions(-DBOOST_ASIO_STANDALONE)
 
 target_include_directories(shadps4 PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
 
+target_include_directories(shadps4 PRIVATE /usr/include/half)
+
 # Shaders sources
 set(HOST_SHADERS_INCLUDE ${CMAKE_CURRENT_SOURCE_DIR}/src/video_core/host_shaders)
 
