diff --git a/externals/CMakeLists.txt b/externals/CMakeLists.txt
index eb3723f2..5dcfa7a3 100644
--- a/externals/CMakeLists.txt
+++ b/externals/CMakeLists.txt
@@ -259,8 +259,8 @@ add_subdirectory(ext-wepoll)
 endif()
 
 #nlohmann json 
-set(JSON_BuildTests OFF CACHE INTERNAL "")
-add_subdirectory(json)
+# set(JSON_BuildTests OFF CACHE INTERNAL "")
+# add_subdirectory(json)
 
 # miniz
-add_subdirectory(miniz)
+find_package(miniz REQUIRED)
