diff --git a/externals/CMakeLists.txt b/externals/CMakeLists.txt
index 4350948b..54773181 100644
--- a/externals/CMakeLists.txt
+++ b/externals/CMakeLists.txt
@@ -194,17 +194,14 @@ if (CMAKE_BUILD_TYPE STREQUAL "Release")
 else()
     option(TRACY_ENABLE "" ON)
 endif()
-option(TRACY_NO_CRASH_HANDLER "" ON) # Otherwise texture cache exceptions will be treaten as a crash
-option(TRACY_ON_DEMAND "" ON)
-option(TRACY_NO_FRAME_IMAGE "" ON)
-option(TRACY_FIBERS "" OFF) # For AmdGpu frontend profiling, disabled due to instability
-option(TRACY_NO_SYSTEM_TRACING "" ON)
-option(TRACY_NO_CALLSTACK "" ON)
-option(TRACY_NO_CODE_TRANSFER "" ON)
-option(TRACY_NO_SAMPLING "" ON)
-option(TRACY_ONLY_LOCALHOST "" ON)
-option(TRACY_NO_CONTEXT_SWITCH "" ON)
-add_subdirectory(tracy)
+find_package(PkgConfig QUIET)
+pkg_search_module(TRACY REQUIRED tracy)
+add_library(TracyClient INTERFACE)
+target_include_directories(TracyClient INTERFACE ${TRACY_INCLUDEDIR})
+target_compile_options(TracyClient INTERFACE ${TRACY_CFLAGS})
+target_link_options(TracyClient INTERFACE ${TRACY_LDFLAGS})
+target_link_libraries(TracyClient INTERFACE ${TRACY_LIBRARIES} ${TRACY_LINK_LIBRARIES})
+add_library(Tracy::TracyClient ALIAS TracyClient)
 
 # pugixml
 if (NOT TARGET pugixml::pugixml)
diff --git a/src/common/debug.h b/src/common/debug.h
index 92bf7255..8562bfbe 100644
--- a/src/common/debug.h
+++ b/src/common/debug.h
@@ -11,7 +11,7 @@
 #error What the fuck is this compiler
 #endif
 
-#include <tracy/Tracy.hpp>
+#include <tracy/tracy/Tracy.hpp>
 
 static inline bool IsProfilerConnected() {
 #if TRACY_ENABLE
diff --git a/src/video_core/renderer_vulkan/vk_instance.h b/src/video_core/renderer_vulkan/vk_instance.h
index 30848e8b..e9238ce7 100644
--- a/src/video_core/renderer_vulkan/vk_instance.h
+++ b/src/video_core/renderer_vulkan/vk_instance.h
@@ -9,7 +9,7 @@
 #include "video_core/renderer_vulkan/vk_platform.h"
 
 #define TRACY_VK_USE_SYMBOL_TABLE
-#include <tracy/TracyVulkan.hpp>
+#include <tracy/tracy/TracyVulkan.hpp>
 
 namespace Frontend {
 class WindowSDL;
