diff --git a/src/qt_gui/gui_context_menus.h b/src/qt_gui/gui_context_menus.h
index d3c51be8..b1bd725f 100644
--- a/src/qt_gui/gui_context_menus.h
+++ b/src/qt_gui/gui_context_menus.h
@@ -116,9 +116,6 @@ public:
 
         compatibilityMenu->addAction(updateCompatibility);
         compatibilityMenu->addAction(viewCompatibilityReport);
-        if (Common::isRelease) {
-            compatibilityMenu->addAction(submitCompatibilityReport);
-        }
 
         menu.addMenu(compatibilityMenu);
 
