diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8050cf4..90333f8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,7 +16,7 @@ include(defaults)
 include(helpers)
 
 add_library(${CMAKE_PROJECT_NAME} MODULE)
-add_subdirectory(deps)
+# add_subdirectory(deps)
 
 string(TIMESTAMP TODAY "%Y.%m.%d %H:%M")
 add_definitions(-DBUILD_TIME="${TODAY}")
@@ -24,8 +24,12 @@ add_definitions(-DTUNA_VERSION="${PROJECT_VERSION}")
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/external")
 
+find_package(PkgConfig REQUIRED)
 find_package(CURL REQUIRED)
 find_package(ZLIB REQUIRED)
+find_package(httplib REQUIRED)
+pkg_check_modules(TAGLIB REQUIRED taglib)
+pkg_check_modules(LIBMPDCLIENT REQUIRED libmpdclient)
 include_directories(${ZLIB_INCLUDE_DIR})
 include_directories(${LIBCURL_INCLUDE_DIRS})
 include_directories(${TAGLIB_INCLUDE_DIRS})
@@ -162,7 +166,7 @@ target_link_libraries(${CMAKE_PROJECT_NAME} PRIVATE OBS::libobs)
 
 
 find_package(Qt6 COMPONENTS Widgets Core)
-target_link_libraries(${CMAKE_PROJECT_NAME} PRIVATE Qt6::Core Qt6::Widgets tag mpdclient ZLIB::ZLIB CURL::libcurl)
+target_link_libraries(${CMAKE_PROJECT_NAME} PRIVATE Qt6::Core Qt6::Widgets ${TAGLIB_LIBRARIES} ${LIBMPDCLIENT_LIBRARIES} ZLIB::ZLIB CURL::libcurl httplib::httplib)
 set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES AUTOMOC ON AUTOUIC ON AUTORCC ON)
 
 configure_file(src/plugin-macros.h.in ${CMAKE_SOURCE_DIR}/src/plugin-macros.generated.h)
