diff --git a/pyproject.toml b/pyproject.toml
index e6374bf..356cefb 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -6,7 +6,6 @@ authors = ["GGML <ggml@ggml.ai>"]
 packages = [
     {include = "gguf"},
     {include = "gguf/py.typed"},
-    {include = "scripts"},
 ]
 readme = "README.md"
 homepage = "https://ggml.ai"
@@ -30,6 +29,6 @@ requires = ["poetry-core>=1.0.0"]
 build-backend = "poetry.core.masonry.api"
 
 [tool.poetry.scripts]
-gguf-convert-endian = "scripts:gguf_convert_endian_entrypoint"
-gguf-dump = "scripts:gguf_dump_entrypoint"
-gguf-set-metadata = "scripts:gguf_set_metadata_entrypoint"
+gguf-convert-endian = "gguf.scripts:gguf_convert_endian_entrypoint"
+gguf-dump = "gguf.scripts:gguf_dump_entrypoint"
+gguf-set-metadata = "gguf.scripts:gguf_set_metadata_entrypoint"
diff --git a/scripts/__init__.py b/scripts/__init__.py
index 77132db..6e7f35b 100644
--- a/scripts/__init__.py
+++ b/scripts/__init__.py
@@ -5,8 +5,8 @@ from importlib import import_module
 
 os.environ["NO_LOCAL_GGUF"] = "TRUE"
 
-gguf_convert_endian_entrypoint = import_module("scripts.gguf-convert-endian").main
-gguf_dump_entrypoint           = import_module("scripts.gguf-dump").main
-gguf_set_metadata_entrypoint   = import_module("scripts.gguf-set-metadata").main
+gguf_convert_endian_entrypoint = import_module("gguf.scripts.gguf-convert-endian").main
+gguf_dump_entrypoint           = import_module("gguf.scripts.gguf-dump").main
+gguf_set_metadata_entrypoint   = import_module("gguf.scripts.gguf-set-metadata").main
 
 del import_module, os
