diff --git a/pyproject.toml b/pyproject.toml
index 5ef013279..d11c5f75f 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -40,14 +40,7 @@ classifiers = [
 ]
 dynamic = ["version"]
 dependencies = [
-    "brotli; implementation_name=='cpython'",
-    "brotlicffi; implementation_name!='cpython'",
-    "certifi",
-    "mutagen",
-    "pycryptodomex",
-    "requests>=2.31.0,<3",
-    "urllib3>=1.26.17,<3",
-    "websockets>=12.0",
+    "pycryptodome",
 ]
 
 [project.optional-dependencies]
