diff --git a/devices/CMakeLists.txt b/devices/CMakeLists.txt
index 4e25cf2..accbba7 100644
--- a/devices/CMakeLists.txt
+++ b/devices/CMakeLists.txt
@@ -137,11 +137,16 @@ if(OIDN_DEVICE_HIP)
     CMAKE_CACHE_ARGS
       -DCMAKE_PREFIX_PATH:STRING=${_hip_prefix_path_str}
       -DCMAKE_CXX_COMPILER:FILEPATH=${OIDN_DEVICE_HIP_COMPILER}
-      -DCMAKE_TOOLCHAIN_FILE:FILEPATH=${CMAKE_TOOLCHAIN_FILE}
       -DCMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE}
       -DCMAKE_INSTALL_PREFIX:PATH=${CMAKE_CURRENT_BINARY_DIR}/hip/preinstall
       -DCMAKE_INSTALL_BINDIR:PATH=${CMAKE_INSTALL_BINDIR}
       -DCMAKE_INSTALL_LIBDIR:PATH=${CMAKE_INSTALL_LIBDIR}
+      -DCMAKE_C_FLAGS:STRING=-fcf-protection=none
+      -DCMAKE_CXX_FLAGS:STRING=-fcf-protection=none
+      -DCMAKE_LD_FLAGS:STRING=-fcf-protection=none
+      -DGPU_TARGETS:STRING=${AMDGPU_TARGETS}
+      -DAMDGPU_TARGETS:STRING=${AMDGPU_TARGETS}
+      -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON
       -DOIDN_ROOT_BINARY_DIR:PATH=${OIDN_ROOT_BINARY_DIR}
       -DOIDN_INSTALL_RPATH_PREFIX:PATH=${CMAKE_INSTALL_PREFIX}
       -DOIDN_INSTALL_DEPENDENCIES:BOOL=${OIDN_INSTALL_DEPENDENCIES}
diff --git a/devices/hip/CMakeLists.txt b/devices/hip/CMakeLists.txt
index 2bfacc2..8e70fe8 100644
--- a/devices/hip/CMakeLists.txt
+++ b/devices/hip/CMakeLists.txt
@@ -25,8 +25,8 @@ include(oidn_common_external)
 # FIXME: The HIP runtime has a bug that causes a crash if the kernels are not compiled for all
 # targets detected in the system. As a workaround, we compile for more targets then we actually
 # support to avoid the crash on systems with dedicated + integrated GPUs.
-set(GPU_TARGETS "gfx1030,gfx1031,gfx1032,gfx1033,gfx1034,gfx1035,gfx1036,gfx1100,gfx1101,gfx1102,gfx1103" CACHE INTERNAL "")
-set(AMDGPU_TARGETS ${GPU_TARGETS} CACHE INTERNAL "")
+set(GPU_TARGETS "gfx1030,gfx1031,gfx1032,gfx1033,gfx1034,gfx1035,gfx1036,gfx1100,gfx1101,gfx1102,gfx1103" CACHE STRING "")
+set(AMDGPU_TARGETS ${GPU_TARGETS} CACHE STRING "")
 
 # Find HIP
 find_package(hip REQUIRED)
