diff --git a/devices/hip/hip_device.cpp b/devices/hip/hip_device.cpp
index e3ea524..d384119 100644
--- a/devices/hip/hip_device.cpp
+++ b/devices/hip/hip_device.cpp
@@ -196,7 +196,7 @@ OIDN_NAMESPACE_BEGIN
     if (hipPointerGetAttributes(&attrib, ptr) != hipSuccess)
       return Storage::Undefined;
 
-    switch (attrib.memoryType)
+    switch (attrib.type)
     {
     case hipMemoryTypeHost:
       return Storage::Host;
