diff --git a/CMakeLists.txt b/CMakeLists.txt
index a2020e9..92c91aa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,7 +11,7 @@ endif()
 add_compile_options(-fPIC
     -Wno-deprecated-declarations
     -Wno-unused-template)
-
+find_package(volk CONFIG)
 add_subdirectory(framegen)
 
 if(LSFGVK_EXCESS_DEBUG)
diff --git a/framegen/CMakeLists.txt b/framegen/CMakeLists.txt
index 9ac0787..c03817d 100644
--- a/framegen/CMakeLists.txt
+++ b/framegen/CMakeLists.txt
@@ -29,7 +29,6 @@ file(GLOB SOURCES
     "v3.1p_src/shaders/*.cpp"
     "v3.1p_src/utils/*.cpp"
     "v3.1p_src/*.cpp"
-    "src/thirdparty/*.c"
 )
 
 add_library(lsfg-vk-framegen STATIC ${SOURCES})
@@ -38,14 +37,12 @@ add_library(lsfg-vk-framegen STATIC ${SOURCES})
 set_target_properties(lsfg-vk-framegen PROPERTIES
     CXX_STANDARD 20
     CXX_STANDARD_REQUIRED ON)
-target_include_directories(lsfg-vk-framegen SYSTEM
-    PUBLIC include/thirdparty)
 target_include_directories(lsfg-vk-framegen
     PUBLIC include
     PUBLIC public
     PRIVATE v3.1_include
     PRIVATE v3.1p_include)
-
+target_link_libraries(lsfg-vk-framegen PUBLIC volk)
 # diagnostics
 if(CMAKE_BUILD_TYPE STREQUAL "Debug")
     set_target_properties(lsfg-vk-framegen PROPERTIES
