diff --git a/src/ArchiveManagement/NexusMods.FileExtractor/Extractors/SevenZipExtractor.cs b/src/ArchiveManagement/NexusMods.FileExtractor/Extractors/SevenZipExtractor.cs
index 4ddc23b8..85c8f33e 100644
--- a/src/ArchiveManagement/NexusMods.FileExtractor/Extractors/SevenZipExtractor.cs
+++ b/src/ArchiveManagement/NexusMods.FileExtractor/Extractors/SevenZipExtractor.cs
@@ -227,12 +227,7 @@ private static string GetExtractorExecutable(IFileSystem fileSystem)
     /// <remarks>
     /// See https://github.com/Nexus-Mods/NexusMods.App/issues/1306 for details.
     /// </remarks>
-    private const bool UseSystemExtractor =
-#if NEXUSMODS_APP_USE_SYSTEM_EXTRACTOR
-        true;
-#else
-        false;
-#endif
+    private const bool UseSystemExtractor = true;
     
     private static string FindSystem7zOnLinux()
     {
