diff --git a/examples/exr2ldr/exr2ldr.cc b/examples/exr2ldr/exr2ldr.cc
index a401211..8a4aa73 100644
--- a/examples/exr2ldr/exr2ldr.cc
+++ b/examples/exr2ldr/exr2ldr.cc
@@ -8,7 +8,7 @@
 #endif
 
 #define STB_IMAGE_RESIZE_IMPLEMENTATION
-#include "stb_image_resize.h"
+#include "deprecated/stb_image_resize.h"
 
 #define STB_IMAGE_WRITE_IMPLEMENTATION
 #include "stb_image_write.h"
diff --git a/examples/nornalmap/main.cc b/examples/nornalmap/main.cc
index 6571739..41bb6a2 100644
--- a/examples/nornalmap/main.cc
+++ b/examples/nornalmap/main.cc
@@ -7,7 +7,7 @@
 #endif
 
 #define STB_IMAGE_RESIZE_IMPLEMENTATION
-#include "stb_image_resize.h"
+#include "deprecated/stb_image_resize.h"
 
 #define STB_IMAGE_WRITE_IMPLEMENTATION
 #include "stb_image_write.h"
diff --git a/examples/resize/resize.cc b/examples/resize/resize.cc
index c78d9a8..e18c374 100644
--- a/examples/resize/resize.cc
+++ b/examples/resize/resize.cc
@@ -3,7 +3,7 @@
 #include <vector>
 
 #define STB_IMAGE_RESIZE_IMPLEMENTATION
-#include "stb_image_resize.h"
+#include "deprecated/stb_image_resize.h"
 
 #include "tinyexr.h"
 
