diff --git a/examples/exr2ldr/exr2ldr.cc b/examples/exr2ldr/exr2ldr.cc
index a401211..8a4aa73 100644
--- a/examples/exr2ldr/exr2ldr.cc
+++ b/examples/exr2ldr/exr2ldr.cc
@@ -8,7 +8,7 @@
 #endif
 
 #define STB_IMAGE_RESIZE_IMPLEMENTATION
-#include "stb_image_resize.h"
+#include "deprecated/stb_image_resize.h"
 
 #define STB_IMAGE_WRITE_IMPLEMENTATION
 #include "stb_image_write.h"
diff --git a/examples/exrfilter/exrfilter.cc b/examples/exrfilter/exrfilter.cc
index b766c1d..cbb7727 100644
--- a/examples/exrfilter/exrfilter.cc
+++ b/examples/exrfilter/exrfilter.cc
@@ -1,6 +1,7 @@
 #include <cstdio>
 #include <cstdlib>
 #include <vector>
+#include <iostream>
 
 #include "tinyexr.h"
 
diff --git a/examples/nornalmap/main.cc b/examples/nornalmap/main.cc
index 6571739..41bb6a2 100644
--- a/examples/nornalmap/main.cc
+++ b/examples/nornalmap/main.cc
@@ -7,7 +7,7 @@
 #endif
 
 #define STB_IMAGE_RESIZE_IMPLEMENTATION
-#include "stb_image_resize.h"
+#include "deprecated/stb_image_resize.h"
 
 #define STB_IMAGE_WRITE_IMPLEMENTATION
 #include "stb_image_write.h"
@@ -18,6 +18,8 @@
 #pragma clang diagnostic pop
 #endif
 
+#include <iostream>
+
 namespace {
 
 static void vnormalize(float v[3]) {
diff --git a/examples/resize/resize.cc b/examples/resize/resize.cc
index c78d9a8..e18c374 100644
--- a/examples/resize/resize.cc
+++ b/examples/resize/resize.cc
@@ -3,7 +3,7 @@
 #include <vector>
 
 #define STB_IMAGE_RESIZE_IMPLEMENTATION
-#include "stb_image_resize.h"
+#include "deprecated/stb_image_resize.h"
 
 #include "tinyexr.h"
 
