diff --git a/apps/app-frontend/src/App.vue b/apps/app-frontend/src/App.vue
index 69483075..0b8f4827 100644
--- a/apps/app-frontend/src/App.vue
+++ b/apps/app-frontend/src/App.vue
@@ -179,18 +179,7 @@ async function setupApp() {
     }),
   )
 
-  useFetch(
-    `https://api.modrinth.com/appCriticalAnnouncement.json?version=${version}`,
-    'criticalAnnouncements',
-    true,
-  ).then((res) => {
-    if (res && res.header && res.body) {
-      criticalErrorMessage.value = res
-    }
-  })
-
   get_opening_command().then(handleCommand)
-  checkUpdates()
   fetchCredentials()
 }
 
@@ -308,15 +297,6 @@ async function handleCommand(e) {
 
 const updateAvailable = ref(false)
 async function checkUpdates() {
-  const update = await check()
-  updateAvailable.value = !!update
-
-  setTimeout(
-    () => {
-      checkUpdates()
-    },
-    5 * 1000 * 60,
-  )
 }
 
 function handleClick(e) {
