diff --git a/apps/app-frontend/src/App.vue b/apps/app-frontend/src/App.vue
index 786e1f31..d8e2ed07 100644
--- a/apps/app-frontend/src/App.vue
+++ b/apps/app-frontend/src/App.vue
@@ -162,15 +162,7 @@ async function setupApp() {
     }),
   )
 
-  useFetch(
-    `https://api.modrinth.com/appCriticalAnnouncement.json?version=${version}`,
-    'criticalAnnouncements',
-    true,
-  ).then((res) => {
-    if (res && res.header && res.body) {
-      criticalErrorMessage.value = res
-    }
-  })
+
 
   useFetch(`https://modrinth.com/blog/news.json`, 'news', true).then((res) => {
     if (res && res.articles) {
@@ -179,7 +171,6 @@ async function setupApp() {
   })
 
   get_opening_command().then(handleCommand)
-  checkUpdates()
   fetchCredentials()
 }
 
@@ -309,15 +300,7 @@ async function handleCommand(e) {
 
 const updateAvailable = ref(false)
 async function checkUpdates() {
-  const update = await check()
-  updateAvailable.value = !!update
-
-  setTimeout(
-    () => {
-      checkUpdates()
-    },
-    5 * 1000 * 60,
-  )
+
 }
 
 function handleClick(e) {
