diff --git a/app/node/project/serializer/serializer230220.cpp b/app/node/project/serializer/serializer230220.cpp
index 06e140ad6..1705a5802 100644
--- a/app/node/project/serializer/serializer230220.cpp
+++ b/app/node/project/serializer/serializer230220.cpp
@@ -190,9 +190,9 @@ ProjectSerializer230220::LoadData ProjectSerializer230220::Load(Project *project
             } else if (attr.name() == QStringLiteral("ptr")) {
               ptr = attr.value().toULongLong();
             } else if (attr.name() == QStringLiteral("items")) {
-              QVector<QStringRef> l = attr.value().split(',');
+              QVector<QStringView> l = attr.value().split(',');
               items.reserve(l.size());
-              for (const QStringRef &s : l) {
+              for (const QStringView &s : l) {
                 items.append(s.toULongLong());
               }
             }
