diff --git a/app/render/renderer.cpp b/app/render/renderer.cpp
index d00fa0d40..840dfcb12 100644
--- a/app/render/renderer.cpp
+++ b/app/render/renderer.cpp
@@ -240,9 +240,16 @@ bool Renderer::GetColorContext(const ColorTransformJob &color_job, Renderer::Col
       const char* sampler_name = nullptr;
       unsigned int width = 0, height = 0;
       OCIO::GpuShaderDesc::TextureType channel = OCIO::GpuShaderDesc::TEXTURE_RGB_CHANNEL;
+#if OCIO_VERSION_HEX >= 0x02030000
+      OCIO::GpuShaderDesc::TextureDimensions dimensions = OCIO::GpuShaderDesc::TEXTURE_2D;
+#endif
       OCIO::Interpolation interpolation = OCIO::INTERP_LINEAR;
 
+#if OCIO_VERSION_HEX >= 0x02030000
+      shader_desc->getTexture(i, tex_name, sampler_name, width, height, channel, dimensions, interpolation);
+#else
       shader_desc->getTexture(i, tex_name, sampler_name, width, height, channel, interpolation);
+#endif
 
       if (!tex_name || !*tex_name
           || !sampler_name || !*sampler_name
